### ISSUE TITLE
Symbol's function definition is void: entities-adjacencies

### ISSUE THREAD

**Issue by ltrojan** [org-trello/org-trello#378]
## Seems to be a bug...

1. trello key installed correctly (had an issue as email was used instead of the username...)
2. create a new org file for a trello board
3. Download board property
4. run `org-trello-sync-buffer`
5. emacs doesn't sync

in the status mini-buffer, emacs prints the following

```
Symbol's function definition is void: entities-adjacencies
```

in the `*Messages*` buffer:

```
org-trello - Request 'sync org buffer to trello board'...
Loading /home/ltrojan/.emacs.d/.trello/lorenzotrojan3.el (source)...done
org-trello - Synchronizing org-mode file to board 'Dev Team'. This may take some time, some coffee may be a good idea...
orgtrello-controller--map-cards-to-computations: Symbol's function definition is void: entities-adjacencies
```

### M-x org-trello-bug-report

```
- A scrambled sample (of the user's and board's ids) of your org-trello buffer with problems.
- Report the following message trace inside your issue.

System information:
- system-type: gnu/linux
- locale-coding-system: utf-8-unix
- emacs-version: GNU Emacs 24.5.1 (x86_64-pc-linux-gnu, GTK+ Version 3.18.9)
 of 2017-09-20 on lcy01-07, modified by Debian
- org version: 8.2.10
- org-trello version: 0.8.0
- org-trello path: /home/ltrojan/.emacs.d/elpa/org-trello-20170225.1247/org-trello.el
- request-backend: curl
- kill-whole-line: nil
```

### Expected behavior

sync buffer with board (board properties had already been installed)

### Actual behavior

nothing happens...

### Steps to reproduce the behavior

See above

**Comment by ardumont** [comment]
Hello,

I agree it's not supposed to happen.
But this sounds like a setup problem.

How did you install org-trello?

Implementation detail: `entities-adjacencies` is an internal variable used by org-trello and it's initialized.

Cheers,

**Comment by bivanov** [comment]
Had same issue; seems to be caused by having both stable.melpa.org/packages and melpa.org/packages as package archives in init file.

**Comment by ltrojan** [comment]
@ardumont @bivanov  thank's for the responses!

this is in my `.emacs.d/init.el` file:

```
(add-to-list 'package-archives '("melpa" . "http://melpa.org/packages/"))
(add-to-list 'package-archives '("org" . "https://orgmode.org/elpa/") t)
(add-to-list 'package-archives '("melpa" . "http://melpa.milkbox.net/packages/") t)
```

I don't remember exactly but i think I first installed org and org trello, but the current package buffer lists this:

```
org-trello         20170225.1247 installed
```

and the package details are:

```
org-trello is an installed package.

     Status: Installed in `~/.emacs.d/elpa/org-trello-20170225.1247/' (unsigned).
    Archive: n/a
    Version: 20170225.1247
   Requires: request-deferred-0.2.0, deferred-0.4.0, s-1.11.0, dash-functional-2.12.1, dash-2.12.1
    Summary: Minor mode to synchronize org-mode buffer and trello board
    Other versions: 20180310.2352 (melpa), 20180310.2352 (melpa).
```

hope this helps...

is there a way I can set that variable up manually?

**Comment by ltrojan** [comment]
Any update on this? ;(

**Comment by ardumont** [comment]
> is there a way I can set that variable up manually?

No. It's not a variable a user can set.
Like i said before, it's an implementation detail you are not supposed to be aware of **iif everything is installed correctly**. This does not look like it is though.

> Any update on this? ;(

Well, looks like the only way forward is to try and install either the same version (if you can) or a latest version ([0.8.1](https://stable.melpa.org/#/org-trello)).

And possibly report back the **\*Compilation\*** buffer's install log after that (if something is still wrong).

Cheers,

**Comment by zhenwenc** [comment]
Having the same issue. I am installing the package with [straight](https://github.com/raxod502/straight.el/issues) and `use-package`.

**Comment by oivvio** [comment]
I'm having the same issue. Here's the output of org-trello-bug-report

```
System information:
- system-type: gnu/linux
- locale-coding-system: utf-8-unix
- emacs-version: GNU Emacs 25.3.1 (x86_64-pc-linux-gnu)
 of 2017-09-12
- org version: 8.2.10
- org-trello version: 0.8.0
- org-trello path: /home/oivvio/.emacs.d/elpa/org-trello-20180331.631/org-trello.el
- request-backend: curl
- kill-whole-line: nil
```

**Comment by daniel-birket** [comment]
I think that I ran into the issue you're describing with void functions. The void functions all used (lexical-let ...), which is defined in cl.el and is not necessarily loaded by a user's init. I added "(eval-when-compile (require 'cl)) ;; lexical-let" above the other (require ...) lines in all the .el files that grep'ed with "lexical-let", then recompiled the directory. This restored the broken sync function. The impacted files are:
org-trello-buffer, org-trello-controller, org-trello-proxy.
I should submit a pull-request...
